okhttp/src/main/java/com/squareup/okhttp/Connection.java

```diff
@@ -444,25 +447,17 @@ boolean isFramed() {
   }
 
   /**
-   * Returns the protocol negotiated by this connection, or {@link
-   * Protocol#HTTP_1_1} if no protocol has been negotiated.
+   * Returns the protocol negotiated by this connection, or {@link Protocol#HTTP_1_1} if no protocol
+   * has been negotiated. This method returns {@link Protocol#HTTP_1_1} even if the remote peer is
+   * using {@link Protocol#HTTP_1_0}.
    */
   public Protocol getProtocol() {
-    return protocol;
-  }
-
-  /**
-   * Sets the protocol negotiated by this connection. Typically this is used
-   * when an HTTP/1.1 request is sent and an HTTP/1.0 response is received.
-   */
-  void setProtocol(Protocol protocol) {
-    if (protocol == null) throw new IllegalArgumentException("protocol == null");
-    this.protocol = protocol;
+    return protocol != null ? protocol : Protocol.HTTP_1_1;
   }
 
   void setTimeouts(int readTimeoutMillis, int writeTimeoutMillis)
       throws RouteException {
-    if (!connected) throw new IllegalStateException("setTimeouts - not connected");
+    if (protocol == null) throw new IllegalStateException("setTimeouts - not connected");
```

Comment:
unrelated: What's up with this exception message? The trace will have the method name.